stl/inc/experimental/filesystem

```diff
@@ -1448,7 +1448,12 @@ private:
     mutable file_status _Mysymstat;
 };
 
-template <class _Prefix_directory>
+enum class _Prefix_directory_t : bool {
+    _Dont_add,
+    _Add,
+};
+
+template <_Prefix_directory_t _Prefix_directory_v>
 class _Directory_iterator {

```

Comment:
This breaks ABI by changing the True Name of a user-visible type:

```
D:\GitHub\STL\out\build\x64>type meow.cpp
```
```cpp
#include <experimental/filesystem>
void user(std::experimental::filesystem::directory_iterator&) {}
```

Before:
```
C:\Temp>cl /EHsc /nologo /W4 /D_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING /c D:\GitHub\STL\out\build\x64\meow.cpp && dumpbin /symbols meow.obj | findstr user
meow.cpp
00C 00000000 SECT4  notype ()    External     | ?user@@YAXAEAV?$_Directory_iterator@U?$integral_constant@_N$00@std@@@v1@filesystem@experimental@std@@@Z (void __cdecl user(class std::experimental::filesystem::v1::_Directory_iterator<struct std::integral_constant<bool,1> > &))
```

After:
```
D:\GitHub\STL\out\build\x64>cl /EHsc /nologo /W4 /D_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING /c meow.cpp && dumpbin /symbols meow.obj | findstr user
meow.cpp
00C 00000000 SECT4  notype ()    External     | ?user@@YAXAEAV?$_Directory_iterator@$00@v1@filesystem@experimental@std@@@Z (void __cdecl user(class std::experimental::filesystem::v1::_Directory_iterator<1> &))
```

Diff:
```diff
-user(class std::experimental::filesystem::v1::_Directory_iterator<struct std::integral_constant<bool,1> > &)
+user(class std::experimental::filesystem::v1::_Directory_iterator<1> &)
```